env/actions.py

```diff
@@ -0,0 +1,74 @@
+from .card import Card
+from dataclasses import dataclass
+from abc import ABC, abstractmethod
+
+
+class Action(ABC):
+
+    @property
+    @abstractmethod
+    def notation(self) -> str:
```

Comment:
Maybe we could use magic methods here

```suggestion
    def __str__(self) -> str:
```